user_management.c

```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+typedef struct {
+    int id;
+    char name[50];
+    int age;
+} User;
+
+void create_user() {
+    FILE* file = fopen("users.txt", "a");
+    if (!file) {
+        printf("Error opening file\n");
+        return;
+    }
+    User user;
+    printf("Enter ID: ");
+    scanf("%d", &user.id);
+    getchar(); 
+    printf("Enter Name: ");
+    fgets(user.name, sizeof(user.name), stdin);
+    user.name[strcspn(user.name, "\n")] = '\0'; 
+    printf("Enter Age: ");
+    scanf("%d", &user.age);
+    fprintf(file, "%d %s %d\n", user.id, user.name, user.age);
+    fclose(file);
+    printf("User added\n");
+}
+
+void read_users() {
+    User user;
+    FILE* file = fopen("users.txt", "r");
```

Comment:
This is repeated in each and every function you can move it to one function so that this not repeated again and again